pkg/registry/core/pod/strategy_test.go

```diff
@@ -1600,3 +1599,128 @@ func TestDropNonEphemeralContainerUpdates(t *testing.T) {
 		})
 	}
 }
+
+func Test_podStrategy_CheckGracefulDelete(t *testing.T) {
+	int64p := func(i int64) *int64 { return &i }
+	tests := []struct {
+		name      string
+		obj       *core.Pod
+		options   func(*metav1.DeleteOptions)
+		wantGrace int64
+	}{
+		{
+			name:      "nil grace period (impossible) is zero",
+			obj:       &core.Pod{Spec: core.PodSpec{NodeName: "test", TerminationGracePeriodSeconds: nil}},
+			wantGrace: 0,
+		},
+		{
+			// Grace period zero has special semantics - it means "delete without waiting for the kubelet".
+			// Since this means the same pod can be created with the same name on a different node, honoring
+			// this value as zero on a pod can result in safety guarantee violations when used incorrectly.
+			// The kubelet already ignores 0 grace period, giving containers an opportunity to shutdown before
+			// force termination, and the apiserver should not allow a zero value default.
+			name:      "nil grace period (impossible) is overridden",
+			obj:       &core.Pod{Spec: core.PodSpec{NodeName: "test", TerminationGracePeriodSeconds: nil}},
+			options:   func(opt *metav1.DeleteOptions) { opt.GracePeriodSeconds = int64p(1) },
+			wantGrace: 1,
+		},
+		{
+			name:      "zero grace period becomes one, ensuring kubelet shutdown is not bypassed",
+			obj:       &core.Pod{Spec: core.PodSpec{NodeName: "test", TerminationGracePeriodSeconds: int64p(0)}},
+			wantGrace: 1,
+		},
+		{
+			name:      "non-zero grace period default is used",
+			obj:       &core.Pod{Spec: core.PodSpec{NodeName: "test", TerminationGracePeriodSeconds: int64p(1)}},
+			wantGrace: 1,
+		},
+		{
+			name:      "large non-zero grace period default is used",
+			obj:       &core.Pod{Spec: core.PodSpec{NodeName: "test", TerminationGracePeriodSeconds: int64p(30)}},
+			wantGrace: 30,
+		},
+
+		{
+			name: "terminal (succeeded) pod is deleted immediately",
+			obj: &core.Pod{
+				Spec:   core.PodSpec{NodeName: "test", TerminationGracePeriodSeconds: int64p(1)},
+				Status: core.PodStatus{Phase: core.PodFailed},
```

Comment:
you mean `core.PodSucceeded` here, right?

Reply:
Yeah cut and paste error